src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -1495,14 +1496,50 @@ private async Task WriteRevertedRevisions(List<Document> list, OperationCancelTo
             list.Clear();
         }
 
-        private bool PrepareRevertedRevisions(DocumentsOperationContext writeCtx, Parameters parameters, RevertResult result, List<Document> list, OperationCancelToken token)
+        private bool PrepareRevertedRevisions(DocumentsOperationContext writeCtx, Parameters parameters, RevertResult result, List<Document> list, OperationCancelToken token, HashSet<string> collections = null)
         {
             using (_database.DocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext readCtx))
             using (readCtx.OpenReadTransaction())
             {
-                var revisions = new Table(RevisionsSchema, readCtx.Transaction.InnerTransaction);
-                foreach (var tvr in revisions.SeekBackwardFrom(RevisionsSchema.FixedSizeIndexes[AllRevisionsEtagsSlice],
-                    parameters.LastScannedEtag))
+                IEnumerable<Table.TableValueHolder> tvrs = null;
+                if (collections != null)
+                {
+                    if (collections.Comparer != null && collections.Comparer != StringComparer.OrdinalIgnoreCase)
+                    {
+                        throw new InvalidOperationException("'collections' hashset must have an 'OrdinalIgnoreCase' comparer");
+                    }
+                    tvrs = Enumerable.Empty<Table.TableValueHolder>();
+                    foreach (var collection in collections)
+                    {
+                        if (collection == null)
+                        {
+                            var msg = "Tried to revert revisions in collection that is null";
+                            if (_logger.IsInfoEnabled)
+                                _logger.Info(msg);
+                            result.WarnAboutFailedCollection(msg);
+                            continue;
+                        }
+                        var collectionName = _documentsStorage.GetCollection(collection, throwIfDoesNotExist: false);
+                        if (collectionName == null)
+                        {
+                            var msg = $"Tried to revert revisions in the collection '{collection}' which does not exist";
+                            if (_logger.IsInfoEnabled)
+                                _logger.Info(msg);
+                            result.WarnAboutFailedCollection(msg);
```

Comment:
can we revert from a collection that does not exist any longer?

Reply:
If the collection is deleted and the revisions are still saved then it will restore the documents,
and if it doesn't exist at all we can log it and continue to the next in the list.
> Better to log / return the message to the user.